### PR TITLE
fix creator hash js fn for bubblegum

### PR DESCRIPTION
**Pull request by ilmoi**
I found that the current function only works for 0 or 1 creators
once you go to 2 or more it fails to generate the correct creator hash, which leads to account_compression::verify_leaf failing

the test currently has an empty array of Creator[] which is why this fell through the cracks, unless I'm missing something ofc

### CODE REVIEW

**Review by danenbm — DISMISSED** [review]

File: bubblegum/js/src/mpl-bubblegum.ts [hunk 1]
```diff
@@ -22,17 +22,16 @@ export function computeDataHash(metadata: MetadataArgs): Buffer {
...
-export function computeCreatorHash(creators: Creator[]): Buffer {
-  let bufferOfCreatorData = Buffer.from([]);
-  let bufferOfCreatorShares = Buffer.from([]);
-  for (const creator of creators) {
-    bufferOfCreatorData = Buffer.concat([
-      bufferOfCreatorData,
-      creator.address.toBuffer(),
-      Buffer.from([creator.share]),
-    ]);
-    bufferOfCreatorShares = Buffer.concat([bufferOfCreatorShares, Buffer.from([creator.verified ? 1 : 0]), Buffer.from([creator.share])]);
```
Comment (lorisleiva) [inline bubblegum/js/src/mpl-bubblegum.ts:34]: Strange, it looks like this variable was never actually used within the function.

**Review by lorisleiva — CHANGES_REQUESTED** [review]
Thanks for that! I'm not too familiar with this repo but it does look like a valid bug fix. Would you be able to add a quick regression test that used to fail before this PR but now passes?

**Comment by danenbm** [comment]
> Thanks for that! I'm not too familiar with this repo but it does look like a valid bug fix. Would you be able to add a quick regression test that used to fail before this PR but now passes?

Seems like we can just add some creators in one of the tests so that it exercises that hashing function in the test.  I tried doing it briefly but having some other issue at the moment and will come back to it unless you add the test @ilmoi before I get to it
```
    let creators: Creator[] = [
      {
        address: payer.publicKey,
        share: 55,
        verified: false,
      },
      {
        address: new Keypair().publicKey,
        share: 45,
        verified: false,
      },
    ]
```

**Comment by danenbm** [comment]
> > Thanks for that! I'm not too familiar with this repo but it does look like a valid bug fix. Would you be able to add a quick regression test that used to fail before this PR but now passes?
> 
> Seems like we can just add some creators in one of the tests so that it exercises that hashing function in the test. I tried doing it briefly but having some other issue at the moment and will come back to it unless you add the test @ilmoi before I get to it
> 
> ```
>     let creators: Creator[] = [
>       {
>         address: payer.publicKey,
>         share: 55,
>         verified: false,
>       },
>       {
>         address: new Keypair().publicKey,
>         share: 45,
>         verified: false,
>       },
>     ]
> ```

I added this code above which exercises the function for 2 creators.  It causes a test failure before this PR's changes, and passes with the commits from this PR.

**Review by lorisleiva — APPROVED** [review]
Thanks Michael!

**Review by samuelvanderwaal — DISMISSED** [review]

**Review by samuelvanderwaal — APPROVED** [review]

**Review by danenbm — APPROVED** [review]
I added test as well.  LGTM

**Pull request merged**